tests/qos/test_qos_sai.py

```diff
@@ -1877,8 +1890,17 @@ def testQosSaiQWatermarkAllPorts(
         except KeyError:
             pytest.skip("Need both TC_TO_PRIORITY_GROUP_MAP and DSCP_TO_TC_MAP and key AZURE to run this test.")
         dscp_to_q_map = {tc_to_dscp_map[tc]:tc_to_q_map[tc] for tc in tc_to_dscp_map}
-        allTestPorts.remove(dutConfig["testPorts"]["src_port_id"])
-        allTestPortIps.remove(dutConfig["testPorts"]["src_port_ip"])
+        try:
+            allTestPorts.remove(dutConfig["testPorts"]["src_port_id"])
+            allTestPortIps.remove(dutConfig["testPorts"]["src_port_ip"])
+        except ValueError:
+            pass
+
+        # Remove the upstream ports from the test port list.
+        for index in range(len(allTestPorts)):
+            if allTestPorts[index] in dutConfig['testPorts']['uplink_port_ids']:
+                allTestPorts.del(index)
```

Comment:
syntax error

Reply:
Fixed the code. Waiting for TB availabiliety to run.